### PR TITLE
Sopan membership pt 60 & pt 61

### PR DESCRIPTION
**Pull request by Sopan-Patil**
```
@ProductDetailsFilterValidation
Feature: Validate Product Filter feature

@ProductDetailsFilterValidation @Smoke @Full
 Scenario: Validate Product Filter feature for Free User

Given Login To Platform
Then Click on Product List and Validate
And logout

```

### CODE REVIEW

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+			CheckFreeMembership_kanjikentei_Productetails();
+		} else if (primeMemberTitle.getText().contentEquals(primeuser)) {
+			productDetails.click();
+			log.info("This is prime user");
+			CheckPrimeMembership_CASEC_Productetails();
+			CheckPrimeMembership_kanjikentei_Productetails();
+		}
+	}
+
+	public void CheckFreeMembership_CASEC_Productetails() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Plz spellcheck wherever needed

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	@FindBy(xpath = "//div[@class='ep-product-list__row']")
+	public WebElement productListCard;
+
+	public void clickProductList() {
+		if (CommonFunctions.waitForVisiblity(productDetails, waitTime)) {
+			productDetails.click();
+		}
+	}
+
+	public void CheckProductFilterFeture() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Please spellCheck

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+					}
+				}
+			}
+		} else {
+			log.info("pagination not exists");
+		}
+
+	}
+
+	public void CheckFreeMembership_kanjikentei_Productetails() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Please refactor to camelCase

File: platform/src/test/java/platformfeatures/PF_ProductDetailsFilterValidation.feature [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@ProductDetailsFilterValidation
+Feature: Validate Product Filter feature
+
+@ProductDetailsFilterValidation @Smoke @Full
```
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_ProductDetailsFilterValidation.feature]: Two similar tags. Please remove any one

**Review by csonparote — CHANGES_REQUESTED** [review]
Please make changes.

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java (hunk 1 above)
Comment (Sopan-Patil) [reply platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Resolved

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java (hunk 3 above)
Comment (Sopan-Patil) [reply platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Resolved

File: platform/src/test/java/platformfeatures/PF_ProductDetailsFilterValidation.feature (hunk 4 above)
Comment (Sopan-Patil) [reply platform/src/test/java/platformfeatures/PF_ProductDetailsFilterValidation.feature]: resolved

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+					}
+				}
+			}
+		} else {
+			log.info("pagination not exists");
+		}
+
+	}
+
+	public void CheckPrimeMembership_CASEC_Productetails() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Please remove _ (underscore) and spell chk

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+					}
+				}
+			}
+		} else {
+			log.info("pagination not exists");
+		}
+
+	}
+
+	public void CheckPrimeMembership_kanjikentei_Productetails() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Please remove _ (underscore) and spell chk

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+			CheckFreeMembershipKanjikenteiProductDetails();
+		} else if (primeMemberTitle.getText().contentEquals(primeuser)) {
+			productDetails.click();
+			log.info("This is prime user");
+			CheckPrimeMembership_CASEC_Productetails();
+			CheckPrimeMembership_kanjikentei_Productetails();
+		}
+	}
+
+	public void CheckFreeMembership_CASEC_ProductDetails() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Please remove _ (underscore) and spell chk

**Review by csonparote — CHANGES_REQUESTED** [review]
PLz make changes

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java (hunk 5 above)
Comment (Sopan-Patil) [reply platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Done

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java (hunk 7 above)
Comment (Sopan-Patil) [reply platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Done

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java (hunk 6 above)
Comment (Sopan-Patil) [reply platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: done

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+					}
+				}
+			}
+		} else {
+			log.info("pagination not exists");
+		}
+
+	}
+
+	public void CheckPrimeMembershipCASECProductetails() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Please spell check

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+					}
+				}
+			}
+		} else {
+			log.info("pagination not exists");
+		}
+
+	}
+
+	public void CheckPrimeMembershipKanjikenteiProductetails() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Please spell check

**Review by csonparote — CHANGES_REQUESTED** [review]
Plz make changes

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java (hunk 8 above)
Comment (Sopan-Patil) [reply platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Done

File: platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java (hunk 9 above)
Comment (Sopan-Patil) [reply platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java]: Done

**Review by csonparote — APPROVED** [review]
Please merge in main

**Pull request merged**